http/server.ts

```diff
@@ -255,17 +255,19 @@ export class Server implements AsyncIterable<ServerRequest> {
 export type HTTPOptions = Omit<Deno.ListenOptions, "transport">;
 
 /**
- * Parse addr from string
+ * Parse addr from string. When only the port is given, the hostname
+ * defaults to 0.0.0.0, except on Windows with 127.0.0.1 as default.
  *
  *     const addr = "::1:8000";
  *     parseAddrFromString(addr);
  *
- * @param addr Address string
+ * @param addr Address string.
  */
 export function _parseAddrFromStr(addr: string): HTTPOptions {
   let url: URL;
   try {
-    const host = addr.startsWith(":") ? `0.0.0.0${addr}` : addr;
+    const defaultAddr = (Deno.build.os === "windows") ? "127.0.0.1" : "0.0.0.0";
```

Comment:
Consider consistency across OS, e.g.

```suggestion
    const defaultAddr = "127.0.0.1";
```

Regardless of the default strategy used, it would be good to note this in the README.md for the http module, and probably also in the jsdocs (for Deno doc) for the public method(s) that call this private function 😊

Reply:
I like this idea more because it's simpler. I've described why in https://github.com/denoland/deno_std/issues/1165